util/memmask/src/memmask.c

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include "os/mynewt.h"
+#include "memmask/memmask.h"
+
+extern int __MYNEWT_SENSITIVE_START, __MYNEWT_SENSITIVE_END;
+
+/**
+ * Clear out sensitive memory area
+ */
+void
+memmask_clear_sensitive_area(void)
+{
+    memset(&__MYNEWT_SENSITIVE_START, 0xff, (&__MYNEWT_SENSITIVE_END-&__MYNEWT_SENSITIVE_START));
```

Comment:
Let's fill it up with a custom pattern and make it configurable.

Reply:
For a 4 byte pattern like `0xbaadbaad` I think we'll have to replace the memset with a 4 byte pointer that loops through each block from `__MYNEWT_SENSITIVE_START` to `__MYNEWT_SENSITIVE_END`.